docs/DeveloperGuide.md

```diff
@@ -1,22 +1,118 @@
+* `getHelp` also guides how the game would be controlled on the CLI.
+* `getHelp` will be overridden by child classes, `TicTacToe` and `Hangman`.
+* ArrayList of `Game` type is created to store multiple instances of `TicTacToe` and `Hangman` to allow many rounds
+of either games to be played.
+* The `isExit` is a `Boolean` variable to be implemented later on, that checks when a user would want to quit an ongoing game or when a game is ended.
+
+### <span style="color:grey;">TicTacToe Implementation:</span>
+The `TicTacToe` class, which inherits from the `Game` class, handles the Tic-Tac-Toe game itself!
+
+The `TicTacToe` class:
+* reads commands using `Scanner`, after which the input is parsed by the `Parser` class. 
+  * Specifically, it takes in values only from `1` to `9`.
+* stores the `board` upon which the game is played on. 
+  * The `board` is a `String` `array` with `9` slots, indicating the possible inputs from the user.
+  * The `board` is also printed at the start of the game and after every move by the user.
+* checks the winner using the `checkWinner(int turnCount)` method, which takes in the number of turns taken so far.
+  * Returns the following based on the state of the game:
+    * `unending` if the game is unfinished or the user inputs `quit`.
+    * `X` if the user wins the game
+    * `O` if the user loses the game
+    * `draw` if the user and the computer draw
+
+
+### <span style="color:grey;">Hangman Implementation:</span>
+The `HangMan` class:
+* When the game is started, `runHangMan` is called.
+* User input is read using `Scanner`. 
+  * Available commands currently are only limited to single characters.
+    * (A future implementation for v2.0 would be to additionally include the ability to input an entire word for guessing)
+* Stores a `wordBank` which contains the list of available words randomly selected for the hangman game.
+* Contains attributes of `chosenWordLength` based on the randomly `chosenWord`.
+* User inputs are compared against the `chosenWord` to check for similar letters used.
+   Each iteration of user input will result in calls of `addGuess`, `printHangMan`, `printLettersGuessed` & `printWordGuesser`.
+  * `addGuess` adds the user input to an arraylist<String> of `allGuessedLetters`
+    * `printLettersGuessed` then iterates through `allGuessedLetters`, printing each element of the array list.
+  * Correctly guessed letters together with `_` blank spaces corresponding to letters not yet guessed, are printed using `printWordGuesser`.
+  * Incorrect guesses however lead to the game `state` increasing by 1.
+    * `printHangMan` prints a different hangman based on the game `state`. 
+    * Should the `state` reach 6, the game is lost.
+* `runHangMan` checks for the number of `_` blank spaces left at each iteration of user input.
+  * Should the number of `_` be 0, the player has won.
```

Comment:
Could include sequence diagrams for sampler user inputs to show how internal interactions lead to game outputs.